transformer_engine/pytorch/ops/op.py

```diff
@@ -232,25 +233,39 @@ def _make_meta(
         )
 
     @classmethod
-    def _maybe_update_fp8_meta(cls, fp8_meta: Optional[dict[str, Any]]) -> None:
+    def _maybe_update_fp8_meta(
+        cls,
+        fp8_meta: Optional[dict[str, Any]],
+        *,
+        fp8_recipe: Optional[DelayedScaling] = None,
+    ) -> None:
         if fp8_meta is None:
             return
 
-        # Update FP8 recipe and communication group
-        recipe = FP8GlobalStateManager.get_fp8_recipe()
-        fp8_meta["recipe"] = recipe
+        # Update FP8 recipe
+        if fp8_recipe is None:
+            fp8_recipe = FP8GlobalStateManager.get_fp8_recipe()
+        if fp8_recipe is None:
+            fp8_recipe = get_default_fp8_recipe()
```

Comment:
Hmmmm, this second if looks like logic that should be inside get_fp8_recipe in the FP8GlobalStateManager.

Reply:
Also, since this is an internal function, couldn't we just always ask for a valid recipe here and just deal with getting it int the caller?

Reply:
This case shouldn't happen in any of our current use-cases (`FP8GlobalStateManager.get_fp8_recipe()` is set within `fp8_autocast`, `fp8_recipe` is provided within `make_graphed_callables`), but it seems delicate to rely on that assumption.